cmd/shell/terminal.go

```diff
@@ -0,0 +1,22 @@
+package shell
+
+import "github.com/moby/term"
+
+// TerminalChecker holds logic to check if environment is a terminal
+type TerminalChecker interface {
+	IsTerminal(interface{}) bool
+}
+
+// DefaultTerminalChecker holds data to check if environment is a terminal
+type DefaultTerminalChecker struct{}
+
+// IsTerminal checks if input is a terminal
+func (t *DefaultTerminalChecker) IsTerminal(in interface{}) (isTerminal bool) {
+	_, isTerminal = term.GetFdInfo(in)
```

Comment:
I think we need to add a `terminal_test.go` file where we write two tests - confirming we are or not in a TTY environment.

On one test, create a pipe, send the writer piece of it down for checking, and asserts it is NOT a valid tty.

On the other test, we create something with PTY (https://github.com/creack/pty):

- We start something (anything - a real command).

```go
        // Create arbitrary command.
        c := exec.Command("bash")

        // Start the command with a pty.
        ptmx, err := pty.Start(c)

       // now we test ptmx (*os.File) for being or not a TTY!
```

https://github.com/creack/pty/blob/a6c0a376f1d083f0ced41ae7bb811497b432f3cd/run.go#L44-L74

- We check the returned `tty` passes the moby check for `isTerminal`.